daemon/daemon.go

```diff
@@ -168,8 +171,14 @@ type Daemon struct {
 	// k8sSvcCache is a cache of all Kubernetes services and endpoints
 	k8sSvcCache k8s.ServiceCache
 
-	mtuConfig     mtu.Configuration
-	policyTrigger *trigger.Trigger
+	mtuConfig      mtu.Configuration
+	policyTrigger  *trigger.Trigger
+	nodeManager    *nodemanager.Manager
```

Comment:
This is stylistic, but it'd be nice to group the `nodeManager`, `nodeConfig`, `nodeRegistrar`, `nodeRegistered`, `localNode`, into a type within the daemon to reduce the amount of fields within the daemon / more logically collocate node-related fields for daemon.

Reply:
I've introduced `nodeDiscovery` type which does this